src/bramble/client/ProjectLimiter.js

```diff
@@ -0,0 +1,69 @@
+define(function() {
+    "use strict";
+    var ProjectLimiter = {};
+
+    var _maxProjectState = {
+            fileSize: 0, // max size of any file in bytes. NOT USED now.
+            fileCount: 0, // total number of files you can have in a project
+            projectSize: 0// total number of bytes for all files in this project
+    };
+
+    var _currentProjectState = {
+        numOfFiles: 0 ,
+        projectSize: 0
+    };
+
+    var _fs;
+
+    ProjectLimiter.initMaxLimits =  function (projectState,fs){
```

Comment:
I'm starting to think that we should separate out the collection of stats from the concept of having limits.  We probably want to end up with 2 modules, one for `ProjectStats` and another for `ProjectLimits`, and the second would use the first to do its work.  I think that could happen in a follow-up PR if you want, and this one could just focus on the `ProjectStats` piece.

Having said that, let's evolve how we do the `init` method.  Since we've already wrapped the filesystem methods we care about, we now need to set our initial state by getting the project root dir that we'll be using for the duration of this instance (e.g., whatever is specified in `Bramble.mount(...)`) and walking that dir recursively, doing `fs.stat` on everything to build an initial picture of what we have.

This step is important because there are different cases that we might encounter on startup:
* The user is starting a brand new project, and there are no files in the filesystem
* The user is opening an existing project, and the filesystem is full of files.

The wrapped filesystem methods, and this code should be reference an internal object that caches various pieces of info (e.g., file size).  After we do this initial walk of the root dir, the wrapped fs methods will capture any changes that happen later on.